tests/rptest/tests/rack_aware_replica_placement_test.py

```diff
@@ -0,0 +1,116 @@
+# Copyright 2022 Vectorized, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+from ducktape.utils.util import wait_until
+from ducktape.mark import matrix
+from rptest.clients.types import TopicSpec
+from rptest.services.cluster import cluster
+from rptest.tests.redpanda_test import RedpandaTest
+from collections import defaultdict
+
+
+class RackAwarePlacementTest(RedpandaTest):
+    def __init__(self, test_context):
+
+        super(RackAwarePlacementTest,
+              self).__init__(test_context=test_context,
+                             num_brokers=6,
+                             extra_rp_conf={'enable_rack_awareness': True})
+
+    def setUp(self):
+        pass
+
+    def _get_partition_count(self, topic):
+        m = self.client().describe_topic(topic)
+        return len(m.partitions)
+
+    def _create_topic(self, topic):
+        """Create topic and wait until all partitions are created.
+        """
+        self.client().create_topic(topic)
+
+        wait_until(
+            lambda: self._get_partition_count(topic.name
+                                              ) == topic.partition_count,
+            timeout_sec=30,
+            backoff_sec=1,
+            err_msg=
+            f"Not all partitions are created in time, expected {topic.partition_count} partitions, found {self._get_partition_count(topic.name)}"
+        )
+
+    def _validate_placement(self, topic, rack_layout, num_replicas):
+        """Validate the replica placement. The method uses provided
+        rack layout and number of replicas for the partitions. 
+        The validation is done by examining existing replica placemnt
+        against the rack layout. The validation succedes if every replica
+        is placed on a different rack or if there is not enough racks on
+        every available rack.
+        """
+        num_racks = len(set(rack_layout))
+        m = self.client().describe_topic(topic.name)
+        replicas = defaultdict(list)
+        for part in m.partitions:
+            for broker in part.replicas:
+                rack = rack_layout[broker - 1]  # broker ids are 1-based
+                replicas[part.id].append(rack)
+
+        for part_id, racks in replicas.items():
+            expected_racks = min(num_racks, num_replicas)
+            self.logger.info(
+                f"partition {part_id} is replicated to racks {racks}, expected {expected_racks} racks, total racks {num_racks}"
+            )
+            assert len(set(racks)) == min(num_racks, num_replicas)
+
+    @cluster(num_nodes=6)
+    @matrix(rack_layout_str=['AAAAAB', 'AABBCC', 'ABCDEF'],
```

Comment:
`aaaa, bbbbb, ccccc` - i don't get the strings.

Reply:
There are 6 brokers and each letter is a rack id. So 'AAAAAB' means that first 5 nodes have rack id = 'A' and the last node has rack = 'B'.

It is possible to pass a list of strings instead of 'AAAAAB' but ducktape adds the parameters of tests with `matrix` decorator to the test name and the test name starts to look extremely ugly. Its also hard to navigate test results in that case.

Reply:
can definitely change them to start with different prefixes

Reply:
Makes sense. Thanks!